sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureResource.java

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.nio;
```

Comment:
If this class is for internal use only would it make sense to move it into `implementation`?

Reply:
I'm not sure. It's pretty integral to some of the functionality and seems like it's of the same kind as the other types in the package. I think I'll leave it here for now, and if it seems it doesn't make sense in the package any more, I can move it later.